Public/Src/Cache/ContentStore/Distributed/NuCache/RocksDbContentLocationDatabase.cs

```diff
@@ -90,13 +91,28 @@ public RocksDbContentLocationDatabase(IClock clock, RocksDbContentLocationDataba
         /// <inheritdoc />
         protected override Task<BoolResult> ShutdownCoreAsync(OperationContext context)
         {
+            lock (this)

```

Comment:
>this [](start = 18, length = 4)

In .net it is highly discourage to ever lock on this or any other publically exposed object (since anybody can lock on it) 

[https://stackoverflow.com/questions/251391/why-is-lockthis-bad](https://stackoverflow.com/questions/251391/why-is-lockthis-bad)

Would you mind adding a local object 
`private object m_lock = new object()`
and lock on that?